fvm/evm/emulator/state/delta.go

```diff
@@ -368,6 +395,31 @@ func (d *DeltaView) SetState(sk types.SlotAddress, value gethCommon.Hash) error
 	return nil
 }
 
+// GetStorageRoot returns some sort of storage root for the given address
+//
+// WARNING! the root that is returned is not a commitment to the state
+// Mostly is returned to satisfy the requirements of the EVM,
+// where the returned value is compared against empty hash and empty root hash
+// values to determine smart contracts that already has data.
+//
+// Here we return values for non-existing accounts, and redirect the call all
+// the way back to the base view. This means that the state root that is returned
+// ignores the updates to slots during the transaction.
+func (d *DeltaView) GetStorageRoot(addr gethCommon.Address) (gethCommon.Hash, error) {
+	exists, err := d.Exist(addr)
+	if err != nil {
+		return gethCommon.Hash{}, err
+	}
+	if !exists {
+		return gethCommon.Hash{}, nil
+	}
```

Comment:
wouldn't this check already be done inside the `GetStorageRoot`

Reply:
right, it doesn't add any value